packages/rss-handler/src/index.ts

```diff
@@ -185,162 +220,182 @@ const getLink = (links: RssFeedItemLink[]) => {
+      continue
+    }
 
-      const fetchResult = await fetchAndChecksum(feedUrl)
-      if (fetchResult.checksum === lastFetchedChecksum) {
-        console.log('feed has not been updated', feedUrl, lastFetchedChecksum)
-        return res.send('ok')
-      }
-      const updatedLastFetchedChecksum = fetchResult.checksum
-
-      // fetch feed
-      let itemCount = 0
-      const feed = await parser.parseString(fetchResult.content)
-      console.log('Fetched feed', feed.title, new Date())
-
-      const feedPubDate = (feed['dc:date'] ||
-        feed.pubDate ||
-        feed.lastBuildDate) as string | undefined
-      console.log('Feed pub date', feedPubDate)
-      if (feedPubDate && new Date(feedPubDate) < new Date(lastFetchedAt)) {
-        console.log('Skipping old feed', feedPubDate)
-        return res.send('ok')
-      }
+    // remember the last item fetched at
+    if (!lastItemFetchedAt || publishedAt > lastItemFetchedAt) {
+      lastItemFetchedAt = publishedAt
+    }
 
-      // save each item in the feed
-      for (const item of feed.items) {
-        // use published or updated if isoDate is not available for atom feeds
-        item.isoDate =
-          item.isoDate ||
-          (item.published as string) ||
-          (item.updated as string) ||
-          (item.created as string)
-        console.log('Processing feed item', item.links, item.isoDate)
-
-        if (!item.links || item.links.length === 0) {
-          console.log('Invalid feed item', item)
-          continue
-        }
+    itemCount = itemCount + 1
+  }
 
-        item.link = getLink(item.links as RssFeedItemLink[])
-        if (!item.link) {
-          console.log('Invalid feed item links', item.links)
-          continue
-        }
+  // no items saved
+  if (!lastItemFetchedAt) {
+    // the feed has been fetched before, no new valid items found
+    if (lastFetchedAt || !lastValidItem) {
+      console.log('No new valid items found')
+      return
+    }
 
-        console.log('Fetching feed item', item.link)
+    // the feed has never been fetched, save at least the last valid item
+    const created = await createSavingItemTask(userId, feedUrl, lastValidItem)
+    if (!created) {
+      console.error('Failed to create task for feed item', lastValidItem.link)
+      throw new Error('Failed to create task for feed item')
+    }
 
-        const publishedAt = item.isoDate ? new Date(item.isoDate) : new Date()
-        // remember the last valid item
-        if (
-          !lastValidItem ||
-          (lastValidItem.isoDate &&
-            publishedAt > new Date(lastValidItem.isoDate))
-        ) {
-          lastValidItem = item
-        }
+    lastItemFetchedAt = lastValidItem.isoDate
+      ? new Date(lastValidItem.isoDate)
+      : new Date()
+  }
 
-        // Max limit per-feed update
-        if (itemCount > 99) {
-          continue
-        }
+  const updateFrequency = (feed['syn:updateFrequency'] ||
```

Comment:
I have a feeling we might need to put this in a function and a try/catch block, because I suspect we get garbage in these values sometimes and we wouldn't want to completely fail if something weird was in updateFrequency

Reply:
Yeah I will put it in a try/catch block